lib/dynamodb/document_client.d.ts

```diff
@@ -12,11 +14,11 @@ interface Blob {}
  * annotates native JavaScript types supplied as input parameters, as well
  * as converts annotated response data to native JavaScript types.
  */
-export class DocumentClient {
+export class DocumentClient<Params extends DocumentClient.ClientParams = {}> {
```

Comment:
You also need to update the typings for S3.ManageUpload (https://github.com/aws/aws-sdk-js/blob/master/lib/s3/managed_upload.d.ts)